docs/architecture.mmd

```diff
@@ -0,0 +1,23 @@
+graph TD
```

Comment:
As the docs structure is getting more complex, there's risk of it turning into a wasteland of random files. I expect thus would feel cleaner if we moved md documents into README.md inside subdirectories along with any supporting files the README needs. So this would become `docs/architecture/`